### PR TITLE
Update HDS models and tests to reflect new Facility Collection

### PR DESCRIPTION
**Pull request by lukoktonos**
JIRA Test: https://jira.mitre.org/browse/BONNIE-968

Related PRs:
- https://github.com/projecttacoma/cql_qdm_patientapi/pull/72
- https://github.com/projecttacoma/bonnie/pull/819
- https://github.com/projectcypress/health-data-standards/pull/530

### CODE REVIEW

File: lib/health-data-standards/models/adverse_event.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 class AdverseEvent < Entry
   field :type, type: Hash
   field :severity, type: Hash
+  field :facility, type: Hash
   # TODO: encounter includes a `facility` object. should that be done here? all we care about is the location.
```
Comment (c-monkey) [inline lib/health-data-standards/models/adverse_event.rb]: this comment is obsolete, should be removed

File: lib/health-data-standards/models/encounter.rb [hunk 2]
```diff
@@ -28,11 +28,24 @@ class Encounter < Entry
...
     
     self.admitTime = (self.admitTime.nil?) ? nil : self.admitTime + date_diff
     self.dischargeTime = (self.dischargeTime.nil?) ? nil : self.dischargeTime + date_diff
   end
 
+  def shift_facility_dates(date_diff)
+    # Facility is now stored as a Hash; shift its dates by updating its start_time and end_time values.
+    if self.facility
+      if self.facility['start_time']
+        self.facility['start_time'] = self.facility['start_time'] + date_diff
```
Comment (c-monkey) [inline lib/health-data-standards/models/encounter.rb]: I think this would be cleaner if you used an assignment operator, i.e. `+=`. See http://www.techotopia.com/index.php/Ruby_Operators#Ruby_Assignment_Operators for description.

Also, this codebase frequently uses trailing expression modifiers so you might rearrange the code that way. I don't feel strongly about this and across the codebase we're inconsistent. See http://ruby-doc.org/core-2.1.2/doc/syntax/control_expressions_rdoc.html#label-Modifier+if+and+unless

So this block might turn into:
```
if self.facility
  self.facility['start_time'] += date_diff if self.facility['start_time']
  self.facility['end_time'] += date_diff if self.facility['end_time']
```

File: lib/health-data-standards/models/encounter.rb [hunk 3]
```diff
@@ -28,11 +28,24 @@ class Encounter < Entry
...
+      end
+
+      if self.facility['end_time']
+        self.facility['end_time'] = self.facility['end_time'] + date_diff
```
Comment (c-monkey) [inline lib/health-data-standards/models/encounter.rb]: ditto

File: lib/health-data-standards/models/encounter.rb [hunk 4]
```diff
@@ -28,11 +28,24 @@ class Encounter < Entry
   def shift_dates(date_diff)
     super
     if self.facility
```
Comment (c-monkey) [inline lib/health-data-standards/models/encounter.rb:30]: this check is now redundant and can be removed - already performed in shift_facility_dates function

**Review by mayerm94 — APPROVED** [review]
Tested, jira test passes

File: test/unit/import/c32/encounter_importer_test.rb [hunk 5]
```diff
@@ -1,4 +1,3 @@
-
 require 'test_helper'
 
 class EncounterImporterTest < Minitest::Test
```
Comment (c-monkey) [inline test/unit/import/c32/encounter_importer_test.rb:3]: you didn't add an assertion for the new field  in facility`type`

File: test/unit/models/date_shift_test.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 require 'test_helper'
+require 'byebug'
```
Comment (c-monkey) [inline test/unit/models/date_shift_test.rb]: please remove this debugger statement

File: test/unit/models/date_shift_test.rb [hunk 7]
```diff
@@ -56,9 +57,9 @@ def test_encounter_shift
 				facility_values.each do |fac_vals|
 					e_vals = vals.merge enc_vals
 			  	entry = Encounter.new(e_vals)
-			  	entry.facility = Facility.new(fac_vals)
+				  entry.facility = Facility.new(fac_vals).as_json()
```
Comment (c-monkey) [inline test/unit/models/date_shift_test.rb:59]: Oh, this file has TABs, no wonder the whitespace is screwy! I hate it. Don't fix that in this PR, though! We should make a separate PR to fix this outlier file (I did a global search, this is the only one). Not a priority for this sprint.

**Review by c-monkey — CHANGES_REQUESTED** [review]

File: test/unit/import/c32/encounter_importer_test.rb (hunk 5 above)
Comment (lukoktonos) [reply test/unit/import/c32/encounter_importer_test.rb:3]: I realized `type` was unused when trying to write this test, so I removed it.

File: lib/health-data-standards/models/encounter.rb (hunk 4 above)
Comment (lukoktonos) [reply lib/health-data-standards/models/encounter.rb:30]: I misread this and removed it from the shift_facility_dates function instead, but in any case there is only 1 check now.

File: test/unit/models/date_shift_test.rb (hunk 7 above)
Comment (lukoktonos) [reply test/unit/models/date_shift_test.rb:59]: Ah, that explains why the indentation seemed so weird.

File: lib/health-data-standards/models/encounter.rb (hunk 4 above)
Comment (c-monkey) [reply lib/health-data-standards/models/encounter.rb:30]: your tweak to remove the empty check here means theoretically one could call shift_facility_dates() with an empty self.facility and cause an error. Why not just move the 3 lines of that new function here and remove it?

**Review by c-monkey — APPROVED** [review]
changes addressed my comments

manual JIRA test 968 passed
